cli/flox-activations/src/cli/fix_paths.rs

```diff
@@ -0,0 +1,216 @@
+use std::collections::{HashSet, VecDeque};
+use std::io::Write;
+use std::path::PathBuf;
+
+use anyhow::bail;
+use clap::Args;
+use log::debug;
+
+use super::{join_dir_list, separate_dir_list};
+
+#[derive(Debug, Args)]
+pub struct FixPathsArgs {
+    #[arg(help = "Which shell syntax to return.")]
+    #[arg(short, long, value_name = "SHELL")]
+    pub shell: String,
+    #[arg(help = "The contents of FLOX_ENV_DIRS.")]
+    #[arg(short, long, value_name = "STRING")]
+    pub env_dirs: String,
+    #[arg(help = "The contents of PATH.")]
+    #[arg(short, long, value_name = "STRING")]
+    pub path: String,
+    #[arg(help = "The contents of MANPATH.")]
+    #[arg(short, long, value_name = "STRING")]
+    pub manpath: String,
+}
+
+impl FixPathsArgs {
+    pub fn handle_inner(&self, output: &mut impl Write) -> Result<(), anyhow::Error> {
+        debug!(
+            "Preparing to fix path vars, FLOX_ENV_DIRS={}",
+            self.env_dirs
+        );
+        let new_path = fix_path_var(&self.env_dirs, &self.path);
+        let new_manpath = fix_manpath_var(&self.env_dirs, &self.manpath);
+        let sourceable_commands = match self.shell.as_ref() {
+            "bash" => {
+                let path_export = format!("export PATH=\"{new_path}\"");
+                let manpath_export = format!("export MANPATH=\"{new_manpath}\"");
+                format!("{path_export};\n{manpath_export};\n")
+            },
+            "zsh" => {
+                let path_export = format!("export PATH=\"{new_path}\"");
+                let manpath_export = format!("export MANPATH=\"{new_manpath}\"");
+                format!("{path_export};\n{manpath_export};\n")
+            },
+            "fish" => {
+                let path_export = format!("set -gx PATH \"{new_path}\"");
+                let manpath_export = format!("set -gx MANPATH \"{new_manpath}\"");
+                format!("{path_export};\n{manpath_export};\n")
+            },
+            "tcsh" => {
+                let path_export = format!("setenv PATH \"{new_path}\"");
+                let manpath_export = format!("setenv MANPATH \"{new_manpath}\"");
```

Comment:
nit: we could factor out the shell specific export syntax between this, set-env-dirs, and activate.rs:activate_in_place

Reply:
PR coming soon for this